### ISSUE TITLE
Backslashes get removed from strings

### ISSUE THREAD

**Issue by StarvingMarvin** [common-workflow-language/cwltool#428]
## Expected Behavior
Tool should print

    print 'Hello World!\n'

## Actual Behavior
result is 

    print 'Hello World!n'

## Workflow Code
```json
{
    "arguments": [],
    "baseCommand": [
        "cat",
        "start.py"
    ],
    "class": "CommandLineTool",
    "cwlVersion": "v1.0",
    "hints": [],
    "inputs": {},
    "label": "dummy3",
    "outputs": {},
    "requirements": [
        {
            "class": "InlineJavascriptRequirement",
        },
        {
            "class": "DockerRequirement",
            "dockerPull": "ubuntu:16.04"
        },
        {
            "class": "InitialWorkDirRequirement",
            "listing": [
                {
                    "entry": "print 'Hello World!\\n'",
                    "entryname": "start.py"
                }
            ]
        }
    ],  
}

```

## Your Environment
* cwltool version:  1.0.20170614025756

**Comment by mr-c** [comment]
Is this a "feature" of YAML?

**Comment by mr-c** [comment]
Recommendation: verify correct behavior according to JSON/YAML specs. Also try `--debug` to see if the change occurs later.

**Comment by tetron** [comment]
Oh, expression evaluation might be screwing around with this.

**Comment by kapilkd13** [comment]
@tetron  agreed. When I was working on windows compatibility, I was testing with `\` as path separator in Javascript expression and it was giving such behaviour. So it might be related to that.

**Comment by mr-c** [comment]
Seems to have been fixed for a long time (works in 3.1.20210823154059)
